core/ui/TiddlerTemplate.tid

```diff
@@ -0,0 +1,7 @@
+title: $:/core/ui/TiddlerTemplate
+
+<$let template={{{ [<currentTiddler>] :cascade[all[shadows+tiddlers]tag[$:/tags/TiddlerTemplateFilter]get[text]] }}}>
+
+<$transclude tiddler={{{ [<template>!is[blank]] ~[[$:/core/ui/ViewTemplate]] }}}/>
```

Comment:
@Jermolene I am curious as to why you prefer the above construction over the following:
```
<$let template={{{ [<currentTiddler>] :cascade[all[shadows+tiddlers]tag[$:/tags/TiddlerTemplateFilter]get[text]] :and[!is[blank]else[$:/core/ui/ViewTemplate]] }}}>

<$transclude tiddler=<<template>>/>
```

Or simplifying even further:
```
<$transclude tiddler={{{ [<currentTiddler>] :cascade[all[shadows+tiddlers]tag[$:/tags/TiddlerTemplateFilter]get[text]] :and[!is[blank]else[$:/core/ui/ViewTemplate]] }}} />
```

If the `:cascade` prefix returned **no** result instead of an empty string when there is no match, the filter could be simplified to:
```
{{{ [<currentTiddler>] :cascade[all[shadows+tiddlers]tag[$:/tags/TiddlerTemplateFilter]get[text]] :else[[$:/core/ui/ViewTemplate]] }}}
```

Reply:
Thanks @saqimtiaz, that's an oversight on my part.

Reply:
(Apologies I pressed send too early)

> I am curious as to why you prefer the above construction over the following:

Thanks @saqimtiaz, that's an oversight on my part.

> If the `:cascade` prefix returned no result instead of an empty string when there is no match, the filter could be simplified to:

The rationale for returning a blank result is that the cascade filter prefix allows multiple input titles to be fed through the cascade in one go. If we returned no result for the input titles that fell through the cascade then it wouldn't be possible to line up the results with the input. Equally, the ability to cascade multiple values at once doesn't seem especially useful...

Reply:
> The rationale for returning a blank result is that the cascade filter prefix allows multiple input titles to be fed through the cascade in one go. If we returned no result for the input titles that fell through the cascade then it wouldn't be possible to line up the results with the input. Equally, the ability to cascade multiple values at once doesn't seem especially useful...

That makes sense, though I am also struggling to think of a use case for cascading multiple values. Users frequently get tripped up by blank values from filters so my thinking was to avoid it if possible. Then again, primitives frequently get put to uses we never could have imagined, so there may well be some utility to being able to pass multiple input titles to a cascade filter run.